File: src/main/java/com/uph/tpsi/services/interfaces/InvoiceService.java
package com.uph.tpsi.services.interfaces;

import com.uph.tpsi.models.Invoice;

import java.util.List;

public interface InvoiceService
{
        List<Invoice> findAll ();

        List<Invoice> findByUser ();

        Invoice create ( Invoice invoice );

        Invoice updateStatus(Long id);

        void deleteById(Long id);
}

File: angular/src/app/services/invoice.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Invoice } from '../invoice';

@Injectable({
  providedIn: 'root'
})
export class InvoiceService {

  url = 'http://localhost:8080/api/invoices/';

  constructor(private http: HttpClient) { }

  findAll() {
    return this.http.get<Invoice[]>(this.url);
  }

  findByUser() {
    return this.http.get<Invoice>(this.url + 'byUser');
  }

  create(invoice: Invoice) {
    return this.http.post<Invoice>(this.url, invoice);
  }

  updateStatus(id: number) {
    return this.http.patch<Invoice>(this.url + id, null);
  }

  deleteById(id: number) {
    this.http.delete(this.url + id).subscribe(res => console.log(res));
  }
}
